<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Minimal Overlaps from BLAST Alignments • epitopefindr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Minimal Overlaps from BLAST Alignments">
<meta property="og:description" content="The purpose of this tool is to describe the alignments among a set of peptide sequences by reporting the overlaps of each peptide's alignments to other peptides in the set. One can imagine inputting a list of peptides enriched by immunoprecipitation to identify corresponding epitopes.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">epitopefindr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.23</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/epitopefindr_avarda_vignette.html">epitopefindr Application: Viral Peptides</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="contents col-md-9">

<div id="epitopefindr-minimal-overlaps-from-blast-alignments" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#epitopefindr-minimal-overlaps-from-blast-alignments" class="anchor"></a>epitopefindr: Minimal Overlaps from BLAST Alignments</h1></div>
<table class="table"><tbody>
<tr class="odd">
<td>Version:</td>
<td>1.1.23</td>
</tr>
<tr class="even">
<td>Date:</td>
<td>March 20, 2019</td>
</tr>
<tr class="odd">
<td>Concept:</td>
<td>Ben Larman, Brandon Sie, Daniel Monaco</td>
</tr>
<tr class="even">
<td>Authors:</td>
<td>Brandon Sie, Athena Chen</td>
</tr>
<tr class="odd">
<td>Contact:</td>
<td>brandonsie at gmail</td>
</tr>
</tbody></table>
</div>
<div id="pipeline-overview" class="section level1">
<h1 class="hasAnchor">
<a href="#pipeline-overview" class="anchor"></a>Pipeline Overview:</h1>
<p>The purpose of this package is to describe the <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> alignments among a set of peptide sequences by reporting the overlaps of each peptide’s alignments to other peptides in the set. One can imagine inputting a list of peptides enriched by immunoprecipitation (e.g. by <a href="https://www.nature.com/articles/s41596-018-0025-6">PhIP-seq</a>) to identify corresponding epitopes.</p>
<p><code>epitopefindr</code> takes a .fasta file listing peptide sequences of interest and calls BLASTp from within R to identify alignments among these peptides. Each peptide’s alignments to other peptides are then simplified to the minimal number of “non overlapping” intervals* of the index peptide that represent all alignments to other peptides reported by BLAST. (*By default, each interval must be at least 7 amino acids long, and two intervals are considered NOT overlapping if they share 6 or fewer amino acids). After the minimal overlaps are identified for each peptide, these overlaps are gathered into aligning groups based on the initial BLAST. For each group, a multiple sequence alignment logo (motif) is generated to represent the collective sequence. Additionally, a spreadsheet is written to list the final trimmed amino acid sequences and some metadata.</p>
<p><img src="https://raw.githubusercontent.com/brandonsie/brandonsie.github.io/master/docs/EpitopeFindRWorkflow2c.png" alt="workflow"></p>
</div>
<div id="setup" class="section level1">
<h1 class="hasAnchor">
<a href="#setup" class="anchor"></a>Setup:</h1>
<ol>
<li>Install <a href="https://www.r-project.org/">R (version 3.5+)</a>.<br>
</li>
<li>Install <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download">BLAST+ (version 2.7.1+)</a>.<br>
</li>
<li>In R console, execute:</li>
</ol>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"></a>
<a class="sourceLine" id="cb1-2" title="2"><span class="co"># Install Bioconductor packages</span></a>
<a class="sourceLine" id="cb1-3" title="3"><span class="cf">if</span> (<span class="op">!</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/ns-load">requireNamespace</a></span>(<span class="st">"BiocManager"</span>)) <span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages</a></span>(<span class="st">"BiocManager"</span>)</a>
<a class="sourceLine" id="cb1-4" title="4">BiocManager<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Biostrings"</span>, <span class="st">"IRanges"</span>, <span class="st">"msa"</span>, <span class="st">"S4Vectors"</span>))</a>
<a class="sourceLine" id="cb1-5" title="5"></a>
<a class="sourceLine" id="cb1-6" title="6"><span class="co"># Install Github packages</span></a>
<a class="sourceLine" id="cb1-7" title="7"><span class="cf">if</span>(<span class="op">!</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/ns-load">requireNamespace</a></span>(<span class="st">"devtools"</span>)) <span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages</a></span>(<span class="st">"devtools"</span>)</a>
<a class="sourceLine" id="cb1-8" title="8">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"mhahsler/rBLAST"</span>)  </a>
<a class="sourceLine" id="cb1-9" title="9">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"brandonsie/epitopefindr"</span>)</a>
<a class="sourceLine" id="cb1-10" title="10"></a>
<a class="sourceLine" id="cb1-11" title="11"><span class="co"># Load &amp; attach</span></a>
<a class="sourceLine" id="cb1-12" title="12"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(epitopefindr)</a></code></pre></div>
<div id="optional-suggested-additional-setup-" class="section level2">
<h2 class="hasAnchor">
<a href="#optional-suggested-additional-setup-" class="anchor"></a>Optional (Suggested) Additional Setup :</h2>
<p><em>(These are not essential to <code>epitopefindr</code>, but are used to generate alignment logo PDFs from the alignment data, which can be valuable visualizations.)</em></p>
<ol>
<li>Install a TeX distribution with <code>pdflatex</code>. (e.g. <a href="https://miktex.org">MiKTeX (version 2.9+)</a>). <em>(Optional; used to convert multiple sequence alignment TeX files to PDF.)</em><br>
</li>
<li>Install <a href="https://www.pdflabs.com/tools/pdftk-server/">pdftk (version 2.02+)</a>. <em>(Optional; used to merge individual PDFs into a single file.)</em>
</li>
</ol>
<hr>
</div>
</div>
<div id="guide" class="section level1">
<h1 class="hasAnchor">
<a href="#guide" class="anchor"></a>Guide</h1>
<ol>
<li>Prepare a list of your peptides of interest using one of the following two methods. Either of these can be fed as the first input parameter to <code>epfind</code>.
<ul>
<li>Make a <a href="https://zhanglab.ccmb.med.umich.edu/FASTA/">FASTA file</a> with peptide names and sequences.</li>
<li>Make an <code>AAStringSet</code> object of peptides (identifier + sequence) as described in the <a href="http://web.mit.edu/~r/current/arch/i386_linux26/lib/R/library/Biostrings/html/XStringSet-class.html">Biostrings documentation</a>.</li>
</ul>
</li>
<li>To run a typical <code>epitopefindr</code> pipeline, try calling <code>epfind</code>:</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"><span class="co"># Basic call</span></a>
<a class="sourceLine" id="cb2-2" title="2"><span class="kw"><a href="reference/epfind.html">epfind</a></span>(<span class="op">&lt;</span>path to .fasta<span class="op">&gt;</span>, <span class="op">&lt;</span>path to output dir<span class="op">&gt;</span>)</a>
<a class="sourceLine" id="cb2-3" title="3"></a>
<a class="sourceLine" id="cb2-4" title="4"><span class="co"># Without pdflatex or pdftk</span></a>
<a class="sourceLine" id="cb2-5" title="5"><span class="kw"><a href="reference/epfind.html">epfind</a></span>(<span class="op">&lt;</span>path to .fasta<span class="op">&gt;</span>, <span class="op">&lt;</span>path to output dir<span class="op">&gt;</span>, </a>
<a class="sourceLine" id="cb2-6" title="6">        <span class="dt">pdflatex =</span> <span class="ot">FALSE</span>, <span class="dt">pdftk =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb2-7" title="7"></a>
<a class="sourceLine" id="cb2-8" title="8"><span class="co"># More stringent e-value threshold</span></a>
<a class="sourceLine" id="cb2-9" title="9"><span class="kw"><a href="reference/epfind.html">epfind</a></span>(<span class="op">&lt;</span>path to .fasta<span class="op">&gt;</span>, <span class="op">&lt;</span>path to output dir<span class="op">&gt;</span>, <span class="dt">e.thresh =</span> <span class="fl">0.0001</span>)</a></code></pre></div>
<p>A brief summary of the functions called by <code>epFind2</code>:</p>
<ul>
<li>
<code>pbCycleBLAST</code> cycles through each input peptide to find the overlap of its alignment with other peptides from the input. Nested within a call to <code>pbCycleBLAST</code> are calls to <code>epitopeBLAST</code>, <code>indexEpitopes</code>.</li>
<li>
<code>trimEpitopes</code> performs a second pass through the identified sequences to tidy alignments.</li>
<li>
<code>indexGroups</code> collects trimmed sequences into aligning groups</li>
<li>
<code>groupMSA</code> creates a multiple sequence alignment motif logo for each group</li>
<li>
<code>outputTable</code> creates a spreadsheet summarizing identified sequences and epitope groups</li>
</ul>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Brandon Sie <br><small class="roles"> Author, maintainer </small>  </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

  </div>
</div>

      <footer><div class="copyright">
  <p>Developed by Brandon Sie.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9000.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
