<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>epitopefindr Application: Viral Peptides • epitopefindr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="epitopefindr Application: Viral Peptides">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">epitopefindr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.23</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/epitopefindr_avarda_vignette.html">epitopefindr Application: Viral Peptides</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/brandonsie/epitopefindr/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li>
  <a href="https://brandonsie.github.io/">brandonsie</a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>epitopefindr Application: Viral Peptides</h1>
                        <h4 class="author">Brandon Sie</h4>
            
            <h4 class="date">2019-03-22</h4>
      
      
      <div class="hidden name"><code>epitopefindr_avarda_vignette.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>This vignette provides a walthrough of using <code>epitopefindr</code> to process an example dataset included with the package, <code><a href="../reference/pairwise_viral_hits.html">epitopefindr::pairwise_viral_hits</a></code>.</p>
<p>The dataset has been previously explored by Monaco et al. <a href="https://www.biorxiv.org/content/biorxiv/early/2018/05/30/333625.full-text.pdf">Biorxiv</a> and describes a set of phage-displayed peptide sequences for which the patient’s antibody specificity demonstrated a statistically significant change when comparing the patient’s data from the <a href="https://www.nature.com/articles/s41596-018-0025-6">PhIP-seq</a> assay.</p>
<p><code>epitopefindr</code> may be useful to look for sequence similarity among these enriched peptides as a step towards identifying critical sequence motifs.</p>
</div>
<div id="loading-the-data" class="section level2">
<h2 class="hasAnchor">
<a href="#loading-the-data" class="anchor"></a>Loading the Data</h2>
<p>We can see that <code>pairwise_viral_hits</code> is a Biostrings::AAStringSet object consisting of the 159 pairwise enriched peptides. These peptides have been annotated with taxonomic and protein name information. This is our input to <code>epitopefindr</code>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(epitopefindr)</a>
<a class="sourceLine" id="cb1-2" title="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(magrittr)</a>
<a class="sourceLine" id="cb1-3" title="3">epitopefindr<span class="op">::</span>pairwise_viral_hits</a></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1">epitopefindr<span class="op">::</span>pairwise_viral_hits</a>
<a class="sourceLine" id="cb2-2" title="2"><span class="co">#&gt;   A AAStringSet instance of length 159</span></a>
<a class="sourceLine" id="cb2-3" title="3"><span class="co">#&gt;       width seq                                        names               </span></a>
<a class="sourceLine" id="cb2-4" title="4"><span class="co">#&gt;   [1]    56 GWTVQAVEVVENFLPVPGST...FAGLIDPSLDQEDFNAVLD Phlebovirus_Sandf...</span></a>
<a class="sourceLine" id="cb2-5" title="5"><span class="co">#&gt;   [2]    56 YPRQMMHPSFAGLIDPSLDQ...SKTINVSLRGAQKRDIEES Phlebovirus_Sandf...</span></a>
<a class="sourceLine" id="cb2-6" title="6"><span class="co">#&gt;   [3]    56 GLRTSGEEVEELEYERLISS...KERGLPYKVRLEKALMSGI Mammarenavirus_Pi...</span></a>
<a class="sourceLine" id="cb2-7" title="7"><span class="co">#&gt;   [4]    56 MADEALYVYLEGPGATLPEQ...PRGVALLSLRLSIIIPRGY Mastadenovirus_Hu...</span></a>
<a class="sourceLine" id="cb2-8" title="8"><span class="co">#&gt;   [5]    56 IGEFSQEGYLLRPRLAKTEL...HTRDQLLSVGDVFDESRMA Parapoxvirus_Pseu...</span></a>
<a class="sourceLine" id="cb2-9" title="9"><span class="co">#&gt;   ...   ... ...</span></a>
<a class="sourceLine" id="cb2-10" title="10"><span class="co">#&gt; [155]    56 MEPRPGASTRRPEGRPQREP...YSEADTEVGGRGDLRPPLT Simplexvirus_Huma...</span></a>
<a class="sourceLine" id="cb2-11" title="11"><span class="co">#&gt; [156]    28 PIIATSDPTPRRDAATKSRRRRPHSRRL               Simplexvirus_Huma...</span></a>
<a class="sourceLine" id="cb2-12" title="12"><span class="co">#&gt; [157]    56 TKKGTSYPKLSKSYTNNKGK...QSLYQNADAYVSVGSSKYN Influenzavirus A_...</span></a>
<a class="sourceLine" id="cb2-13" title="13"><span class="co">#&gt; [158]    56 TKKGTSYPKLSKSYTNNKGK...QTLYQNVDAYVSVGSSKYN Influenzavirus A_...</span></a>
<a class="sourceLine" id="cb2-14" title="14"><span class="co">#&gt; [159]    56 MDVLSKSSLKELLAHLERTP...LISRNEYYNQPYPDVTSLI Seadornavirus_Ban...</span></a>
<a class="sourceLine" id="cb2-15" title="15"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(epitopefindr<span class="op">::</span>pairwise_viral_hits)[<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb2-16" title="16"><span class="co">#&gt; [1] "Phlebovirus_Sandfly fever sicilian virus_Nucleocapsid_1297"</span></a></code></pre></div>
</div>
<div id="run-epitopefindr" class="section level2">
<h2 class="hasAnchor">
<a href="#run-epitopefindr" class="anchor"></a>Run <code>epitopefindr</code>
</h2>
<p>The all-in-one wrapper for running <code>epitopefindr</code> is <code>epfind</code>. At its most basic, <code>epfind</code> takes two input parameters, (1) a set of named peptides, either as an AAStringSet object or as a directory path to a .fasta file that can be read by Biostrings::readAAStringSet, and (2) a path to which output files can be written. More parameters for customization are documented at <code><a href="../reference/epfind.html">?epfind</a></code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" title="1">output.dir &lt;-<span class="st"> "AVARDA_vignette_data/"</span></a>
<a class="sourceLine" id="cb3-2" title="2">msa.path &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(output.dir,<span class="st">"/msa.pdf"</span>)</a>
<a class="sourceLine" id="cb3-3" title="3"></a>
<a class="sourceLine" id="cb3-4" title="4"><span class="cf">if</span>(<span class="op">!</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/files">file.exists</a></span>(msa.path)){</a>
<a class="sourceLine" id="cb3-5" title="5">  epitopefindr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/epitopefindr/topics/epfind">epfind</a></span>(epitopefindr<span class="op">::</span>pairwise_viral_hits, output.dir,</a>
<a class="sourceLine" id="cb3-6" title="6">                       <span class="dt">e.thresh =</span> <span class="fl">0.0001</span>, <span class="dt">min.groupsize =</span> <span class="dv">6</span>)</a>
<a class="sourceLine" id="cb3-7" title="7">  <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlink">unlink</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(output.dir,<span class="st">"intermediate_files"</span>), <span class="dt">recursive =</span> <span class="ot">TRUE</span>, <span class="dt">force =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb3-8" title="8">}</a></code></pre></div>
</div>
<div id="investigate-sequence-alignment-logos" class="section level2">
<h2 class="hasAnchor">
<a href="#investigate-sequence-alignment-logos" class="anchor"></a>Investigate Sequence Alignment Logos</h2>
<p>We can print some of the sequence motifs from this output. Consider MSA 11 as an example. This 18 amino acid motif is represented in 7 of our input peptides. Many of these positions are fully conserved, but positions 3, 5, and 17 are only partially conserved. This illustrates that <code>epitopefindr</code> makes use of BLAST to retain evidence for certain motifs that would be discarded with a gapped k-mer approach.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1"><span class="co">#convert pdf to png</span></a>
<a class="sourceLine" id="cb4-2" title="2">print.range &lt;-<span class="st"> </span><span class="dv">11</span></a>
<a class="sourceLine" id="cb4-3" title="3"></a>
<a class="sourceLine" id="cb4-4" title="4">imgdir &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(output.dir,<span class="st">"/png/"</span>)</a>
<a class="sourceLine" id="cb4-5" title="5"><span class="cf">if</span>(<span class="op">!</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/files2">dir.exists</a></span>(imgdir)) <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/files2">dir.create</a></span>(imgdir)</a>
<a class="sourceLine" id="cb4-6" title="6">imgnames &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(imgdir,<span class="st">"msa-"</span>,print.range,<span class="st">".png"</span>)</a>
<a class="sourceLine" id="cb4-7" title="7"></a>
<a class="sourceLine" id="cb4-8" title="8">pdftools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/pdftools/topics/pdf_render_page">pdf_convert</a></span>(msa.path, <span class="dt">format =</span> <span class="st">"png"</span>, <span class="dt">pages =</span> print.range,</a>
<a class="sourceLine" id="cb4-9" title="9">                      <span class="dt">filenames =</span> imgnames, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<div id="msa-11" class="section level3">
<h3 class="hasAnchor">
<a href="#msa-11" class="anchor"></a>MSA 11</h3>
<p><img src="AVARDA_vignette_data/png/msa-11.png"></p>
</div>
</div>
<div id="network-graph" class="section level2">
<h2 class="hasAnchor">
<a href="#network-graph" class="anchor"></a>Network Graph</h2>
<p>The output of <code>epitopefindr</code> can be fed into network graphing. Each vertex represents a fragment of an input sequence. Each link represents an alignment between two peptide fragments. Group 11 is highlighted in blue.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" title="1">aln &lt;-<span class="st"> </span>data.table<span class="op">::</span><span class="kw">fread</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(output.dir,<span class="st">"/finalAlignments.csv"</span>))</a>
<a class="sourceLine" id="cb5-2" title="2">ver &lt;-<span class="st"> </span>data.table<span class="op">::</span><span class="kw">fread</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(output.dir,<span class="st">"/epitopeSummary.csv"</span>), <span class="dt">header =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb5-3" title="3"></a>
<a class="sourceLine" id="cb5-4" title="4">g11 &lt;-<span class="st"> </span>ver[,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"id"</span>,<span class="st">"11"</span>)]</a>
<a class="sourceLine" id="cb5-5" title="5">g11 &lt;-<span class="st"> </span>g11[<span class="op">!</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">grep</a></span>(<span class="st">"NA$"</span>,g11<span class="op">$</span>id),]</a>
<a class="sourceLine" id="cb5-6" title="6">g11<span class="op">$</span>color &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/ifelse">ifelse</a></span>(g11<span class="op">$</span><span class="st">`</span><span class="dt">11</span><span class="st">`</span> <span class="op">==</span><span class="st"> ""</span>, <span class="st">"red"</span>,<span class="st">"blue"</span>)</a>
<a class="sourceLine" id="cb5-7" title="7"></a>
<a class="sourceLine" id="cb5-8" title="8"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">1</span>)</a>
<a class="sourceLine" id="cb5-9" title="9">g &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/igraph/topics/graph_from_data_frame">graph_from_data_frame</a></span>(aln[,<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>], <span class="dt">directed =</span> <span class="ot">FALSE</span>, <span class="dt">vertices =</span> g11) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-10" title="10"><span class="st">  </span>(igraph<span class="op">::</span>simplify) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(<span class="dt">vertex.label =</span> <span class="ot">NA</span>, <span class="dt">vertex.size =</span> <span class="dv">3</span>, </a>
<a class="sourceLine" id="cb5-11" title="11">                              <span class="dt">arrow.size =</span> <span class="dv">0</span>, <span class="dt">vertex.color =</span> g11<span class="op">$</span>color)</a></code></pre></div>
<p><img src="epitopefindr_avarda_vignette_files/figure-html/igraph-1.png" width="672"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#loading-the-data">Loading the Data</a></li>
      <li><a href="#run-epitopefindr">Run <code>epitopefindr</code></a></li>
      <li><a href="#investigate-sequence-alignment-logos">Investigate Sequence Alignment Logos</a></li>
      <li><a href="#network-graph">Network Graph</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Brandon Sie.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9000.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
